core/drivers/crypto/caam/hal/common/hal_cfg_dt.c

```diff
@@ -98,10 +99,12 @@ void caam_hal_cfg_get_jobring_dt(void *fdt, struct caam_jrcfg *jrcfg)
 
 	jr_offset = find_jr_offset(fdt, DT_STATUS_OK_SEC, &node);
 	if (jr_offset) {
-		/* Disable JR for Normal World */
-		if (dt_enable_secure_status(fdt, node)) {
-			EMSG("Not able to disable JR DTB entry");
-			return;
+		if (!IS_ENABLED(CFG_EMBED_DTB)) {
```

Comment:
` dt_is_rw()` would be a nice helper however here the goal is the change the content of the DTB later passed to normal world, so a helper function like `dt_is_for_ree()`  seems more appropriate.

Reply:
Sure, or maybe even `dt_is_shared_with_nwd()`.

Reply:
Shared DT frightens me a bit. A shared DT should be placed in a safe location for secure world to get safe information from, and be placed in a non-secure memory for normal world to access it. This may requires platform specific means (e.g. using a "secure read/write" and "non-secure read-only" memory) or a dedicated SW sequence (e.g. using secure memory and moving the DTB to non-secure memory once OP-TEE inits are completed). We don't support these things in current OP-TEE implementation. Ideas and proposals are welcome.

Reply:
Yes, it seems more safe to work on a temporary secure copy of the DT which could be copied out again when we're done with it. I don't suppose that's in scope of this PR though.

Reply:
What about `dt_is_embedded()` function as name for function ?

```
bool dt_is_embedded()
{
       return (IS_ENABLED(CFG_EMBED_DTB));
}
```

> Yes, it seems more safe to work on a temporary secure copy of the DT which could be copied out again when we're done with it. I don't suppose that's in scope of this PR though.

Yes, scope of this PR is just to not call `dt_enable_secure_status(`) in case of Embedded DTB

Reply:
> Yes, it seems more safe to work on a temporary secure copy of the DT which could be copied out again when we're done with it.

Secure world cannot rely on a DTB loaded in a memory reacheable from the outer world (from OP-TEE view). For example, if the DTB in loaded in clear in DDR by some early boot stage, REE cannot access it since not booted yet but the DDR could be physically accessed and tampered before OP-TEE boots. In such case its too late for OP-TEE to copy the DT into some secure memory.

> I don't suppose that's in scope of this PR though.

Agree.

> ```
> bool dt_is_embedded()
> {
>        return (IS_ENABLED(CFG_EMBED_DTB));
> }
> ```

```c
bool is_embedded_dt(void *fdt)
{
	return fdt && fdt == get_embedded_dt();
}
```

Reply:
Tried testing with new suggestion on LX2 but facing a crash in board bootup.
Raised an issue for that: https://github.com/OP-TEE/optee_os/issues/5017 
Please have a look.